src/posts/2021-05-03/index.md

```diff
@@ -0,0 +1,662 @@
+* [_Comparing SystemTap and bpftrace_](https://lwn.net/Articles/852112/),
+  <span style="white-space: nowrap;">from Emanuele Rocca.</span>  
+  SystemTap and bpftrace are tools for dynamically instrument the kernel or
+  user applications on Linux, and inspect them in a variety of ways. SystemTap
+  is older and compiles its scripts into kernel modules. Bpftrace uses eBPF
+  programs attached to different probes. This post from LWN.net exposes the
+  differences between the two tools in terms of installation procedure, program
+  structure, and features. As one could expect, using kernel modules is
+  ultimately more powerful, but producing eBPF programs with eBPF is faster and
+  simpler to use.
+
+<!--2021-04-22-->
+* [_Toward signed BPF programs_](https://lwn.net/SubscriberLink/853489/2916fbd97c95c700/),
+  <span style="white-space: nowrap;">from Jonathan Corbet.</span>  
+  Another LWN.net article, this time on a recent proposal sent by Alexei
+  Starovoitov in order to introduce support for signed eBPF programs, so that
+  the system would only accept authorized eBPF programs. The mechanism is
+  compared with the one in place for kernel modules, but signing eBPF programs
+  is more complex (due to the usage of maps, to the relocations steps, to the
+  bytecode changes that the loader applies before injecting the program). The
+  proposal currently under review solves this with a special eBPF program of a
+  new type, which would be allowed to run `bpf()` and `close()` system calls
+  and would load other eBPF programs from within the kernel. The work is still
+  in progress.
+
+<!--2021-03-11-->
+* [_What is vmlinux.h?_](https://www.grant.pizza/blog/vmlinux-header/),
+  <span style="white-space: nowrap;">from Grant Seltzer.</span>  
+  The vmlinux.h header is generated automatically and contains BTF information
+  about the kernel itself. Modern eBPF programs may use this information to
+  access to a number of features,
+  [CO-RE](https://ebpf.io/blog/ebpf-updates-2020-12#did-you-know-co-re) being
+  one of them.
+
+<!--2021-02-18/20-->
+* [DevConf.CZ 2021](https://www.devconf.info/cz/) ([Schedule](https://devconfcz2021.sched.com/))
+  hosted several talks related to eBPF. Sadly I could not find the videos _and_
+  slides for all presentations.
+    * [_eBPF Iterators_](https://devconfcz2021.sched.com/event/gmK5/ebpf-iterators)
+      ([slides](https://static.sched.com/hosted_files/devconfcz2021/60/bpf-iterators.pdf)),
+      <span style="white-space: nowrap;">from Jiri Olsa.</span>  
+      This is possibly the only document on eBPF iterators we have at the
+      moment. Those iterators are hooked on various objects in the kernel (for
+      example, among many others: tasks, eBPF programs or maps, active TCP
+      connections). Pinned to the eBPF virtual file system, they can be dumped
+      (simply with `cat`) to iterate over the selected objects and print a
+      number of information about them. This is very similar, but more
```

Comment:
Maybe we should just remove “a number of” in that case? Hard to understand what it does with the info otherwise?

Reply:
I'm guessing printing it isn't the only possibility?

Reply:
You process the info and then print the result. I guess you could extract it via a map instead of printing it, for other use cases (although you'd have to `cat` on the pinned path to trigger map updates, which looks like a strange workflow?). But as far as I know, this has been used as an improved version of the procfs so far. I'd like to tell that somehow, and not leave the user wondering “OK but _what do we iterate over the objects for?_”.

Reply:
I get printing is an important use case right now, but surely that's not the only thing you can do with iterators. You can also aggregate information, filter some, etc., no? So what about:
```suggestion
      (simply with `cat`) to iterate over the selected objects and gather
      information to print. This is very similar, but more
```

Reply:
Ah right, you mean it wasn't clear from my phrasing that we can process the info before printing it, is that correct? If so your suggestion looks good, we could use `process` instead of `gather` maybe.

Reply:
Ok, let's go with this then:
```suggestion
      (simply with `cat`) to iterate over the selected objects and process
      information to print. This is very similar, but more
```